include/ros_sec_test/attacks/periodic_attack_component.hpp

```diff
@@ -36,23 +36,25 @@ namespace attacks
 class PeriodicAttackComponent : public rclcpp_lifecycle::LifecycleNode
 {
 public:
+  PeriodicAttackComponent();
   explicit PeriodicAttackComponent(std::string node_name);
   PeriodicAttackComponent(const PeriodicAttackComponent &) = delete;
   PeriodicAttackComponent & operator=(const PeriodicAttackComponent &) = delete;
+  virtual ~PeriodicAttackComponent() = default;
 
-  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+  virtual rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
```

Comment:
Ditto, let's try not to modify the class for testing purposes.

https://github.com/google/googletest/blob/master/googlemock/docs/CookBook.md#mocking-nonvirtual-methods